src/Storages/Cache/ExternalDataSourceCache.cpp

```diff
@@ -216,6 +274,17 @@ std::pair<std::unique_ptr<LocalFileHolder>, std::unique_ptr<ReadBuffer>> Externa
             lru_caches->weight());
         return {nullptr, std::move(read_buffer)};
     }
+    /*
+      If read_buffer is seekable, use read_buffer directly inside LocalFileHolder. And once LocalFileHolder is released,
+      start the download process in background.
+      The cache is marked disable until the download process finish.
+      For reading parquet files from hdfs, with this optimization, the speedup can reach 3x.
+    */
```

Comment:
It could work.

But in our test, for the readbuffer read in order, the speed boost is smaller. 
In consideration of reducing the pressure of hdfs, we keep the files in format text/csv read in the old way.